libbpf-rs/src/program.rs

```diff
@@ -418,6 +421,34 @@ impl Program {
         self.as_fd()
     }
 
+    /// Returns program fd by id
+    pub fn get_fd_by_id(id: u32) -> Result<OwnedFd> {
+        let ret = unsafe { libbpf_sys::bpf_prog_get_fd_by_id(id) };
+        let fd = util::parse_ret_i32(ret)?;
+        // SAFETY
+        // A file descriptor coming from the bpf_prog_get_fd_by_id function is always suitable for
+        // ownership and can be cleaned up with close.
+        Ok(unsafe { OwnedFd::from_raw_fd(fd) })
+    }
+
+    /// Returns program id by fd
+    pub fn get_id_by_fd(fd: BorrowedFd) -> Result<u32> {
+        let mut prog_info = libbpf_sys::bpf_prog_info::default();
+        let prog_info_ptr: *mut libbpf_sys::bpf_prog_info = &mut prog_info;
+        let mut len = std::mem::size_of::<libbpf_sys::bpf_prog_info>() as u32;
+        let ret = unsafe {
+            libbpf_sys::bpf_obj_get_info_by_fd(
+                fd.as_raw_fd(),
+                prog_info_ptr as *mut c_void,
+                &mut len,
+            )
+        };
+        if ret != 0 {
+            return Err(Error::Internal(format!("get bpf prog_info error: {}", ret)));
+        }
```

Comment:
What's the reason for not using `parse_ret_*` functionality here? (as we do in `get_fd_by_id`)

Reply:
> What's the reason for not using `parse_ret_*` functionality here? (as we do in `get_fd_by_id`)

I misunderstood the returned value(0 for success and otherwise fail), fixed.